Add high level categories for Accessories and Manufacturers

nhultz opened the pull request:
Solves #30  and #31.

nhultz commented:
@BalestraPatrick Here is an initial pass at adding the high level menus.  Just wanted to get some initial feedback on approach.

The one thing that hasn't been added just yet is highlighting the menu that is clicked just like the other category menus.  I'm also still taking a look to see if I can find some good icons 😄

BalestraPatrick commented on Sources/App/Controllers/ExploreController.swift:
```diff
@@ -25,6 +25,9 @@ final class ExploreController {
         // Get all categories and find out if one of them is selected.
         let categories = try Category.all().select(selected: queryCategory)
 
+        let accessoryCount = categories.reduce(0, {$0 + $1.accessoriesCount})
```
I think we could improve the efficiency of this count. Since at line 42 we query anyway all the accessories, we can directly `count` that array instead of doing a query for each category. Do you agree?

nhultz replied:
Yeah I see what you mean here.  I think I overlooked the part where using the `accessoriesCount` attribute was going to do a query each time 😄 

This doesn't help the case where someone selects a category but we could do the same thing here that you mentioned in your other comment  e.g.  put all accessories into a variable, count them, then filter down to the category that was picked.

Do you think that's better than doing the count of each category?

BalestraPatrick replied:
Yes, that’s exactly what I meant.

BalestraPatrick commented on Sources/App/Controllers/ExploreController.swift:
```diff
@@ -55,12 +60,17 @@ final class ExploreController {
+
+        let categories = try Category.all()
+
+        let accessoryCount = categories.reduce(0, {$0 + $1.accessoriesCount})
```
In this case you can see that at line 63 we still query all the accessories and then we sort them. To avoid doing the `reduce` on all categories here, you could first store the result of `try Accessory.makeQuery().filter("approved", true).all()` in a variable, store the count and then do the `sort` and `filter`.

nhultz replied:
Yup that make sense.  I'll update that handler.

nhultz commented:
I was thinking the same thing about the icons.  The more that I looked at it, the more it looked pretty good without them 😄

nhultz commented:
@BalestraPatrick Alright I think this should be done now.  I updated the count queries to be more performant in the places you discovered and changed the rest of the counts to use the `.count()` function on the query instead of `all().count` which should also speed things up a little bit.

Also, added the new menus to the new Home page.

BalestraPatrick commented:
Hey @nhultz, I will review this and fix the conflicts myself now so that we can merge it! Thanks a lot 👍

BalestraPatrick commented:
Thank you very much @nhultz, your feature is deploying right now 🎉
